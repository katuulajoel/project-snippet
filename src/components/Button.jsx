import PropTypes from "prop-types";
import React from "react";

import { addEventListeners, BUTTON_EVENTS } from "../utils/events";
import { filterButtonProps } from "../utils/forms";

const Button = (props) => {
  return (
    <button
      type={props.type}
      form={props.form}
      className={`btn ${props.variant ? `btn-${props.variant}` : ""} ${
        props.className || ""
      } ${props.block ? "btn-block" : ""} ${
        props.size ? `btn-${props.size}` : ""
      }`}
      {...filterButtonProps(props)}
      {...addEventListeners(BUTTON_EVENTS, props)}
      data-tip={props["data-tip"]}
    >
      {props.children}
    </button>
  );
};

Button.defaultProps = {
  type: "button",
  form: null,
  variant: "default",
  block: false,
};

Button.propTypes = {
  type: PropTypes.string,
  className: PropTypes.string,
  variant: PropTypes.string,
  size: PropTypes.string,
  block: PropTypes.bool,
  children: PropTypes.any,
  form: PropTypes.string,
  "data-tip": PropTypes.string,
};

export default Button;
