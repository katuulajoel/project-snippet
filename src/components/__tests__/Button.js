import React from "react";
import renderer from "react-test-renderer";
import Button from "../Button";

describe("Button test", () => {
  it("Snapshot test for Button component", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
