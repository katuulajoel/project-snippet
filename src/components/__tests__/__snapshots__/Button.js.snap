// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test Snapshot test for Button component 1`] = `
<button
  className="btn btn-default   "
  form={null}
  type="button"
/>
`;
